controllers/grafanacontactpoint_controller.go

```diff
@@ -82,15 +82,13 @@ func (r *GrafanaContactPointReconciler) Reconcile(ctx context.Context, req ctrl.
 	}
 
 	if contactPoint.GetDeletionTimestamp() != nil {
+		// Check if resource needs clean up
 		if controllerutil.ContainsFinalizer(contactPoint, grafanaFinalizer) {
-			err := r.finalize(ctx, contactPoint)
-			if err != nil {
-				return ctrl.Result{RequeueAfter: RequeueDelay}, fmt.Errorf("failed to finalize GrafanaContactPoint: %w", err)
+			if err := r.finalize(ctx, contactPoint); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to finalize GrafanaContactPoint: %w", err)
 			}
-			controllerutil.RemoveFinalizer(contactPoint, grafanaFinalizer)
-			if err := r.Update(ctx, contactPoint); err != nil {
-				r.Log.Error(err, "failed to remove finalizer")
-				return ctrl.Result{RequeueAfter: RequeueDelay}, fmt.Errorf("failed to update GrafanaContactPoint: %w", err)
+			if err := removeFinalizer(ctx, r.Client, contactPoint); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to remove finalizer: %w", err)
```

Comment:
I see that we had some inconsistencies around re-queuing - we were adding `RequeueDelay` for contact points and notification policies (`ctrl.Result{RequeueAfter: RequeueDelay}`), but not for rule groups (`ctrl.Result{}`). In your PR, `ctrl.Result{}`, but I think it's better to have delay everywhere to make sure the operator doesn't cause reconciliation storm. Once that is added to all 3 resources, we can merge the PR.

Reply:
From my testing and @theSuess review comment in: #1796
I believe `requeue` and `requeueDelay` are ignored when the returned error is not nil.
And when testing the error return behaviour, I saw an exponential backoff when an error was repeatedly returned.
I will try and follow up with a screenshot later today.

Reply:
Ah, alright, thanks for clarifying that! I've just injected a test error and saw the exponential backoff, so we should be good to go :)